crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1976,3 +2019,39 @@ pub(crate) fn report_duplicate_bases(
 
     diagnostic.sub(sub_diagnostic);
 }
+
+pub(crate) fn report_invalid_base(
+    context: &InferContext,
+    base_node: &ast::Expr,
+    base_type: Type,
+    class: ClassLiteral,
+) {
+    let db = context.db();
+
+    if base_type.is_assignable_to(db, KnownClass::Type.to_instance(db)) {
```

Comment:
We could also issue `unsupported-base` for types with an `__mro_entries__` member?